design/yaml_file_errors.md

```diff
@@ -0,0 +1,49 @@
+# YAML File Errors
+
+When the output of a rendered file results in invalid YAML, the resource is not included in base and not deployed.
+Additionally, the file name and error is included in the kots.io/v1beta1.Installer yaml (see the example below).
+The error often references a line number in the rendered content which is inaccessible to the user.
+This proposal attempts to make the rendered output available to the user so that it is easier to debug the issue.
+
+## Goals
+
+- Surface the rendered YAML content to the user when unmarshalling fails
+
+## Non Goals
+
+- Attempt to automate fixing any common errors such as unescaped quotes or line breaks
+
+## Design
+
+Add a new directory at the root of the file tree `errors` that will include the raw content of all rendered output for yaml files that failed unmarshalling.
+
+An example installation.yaml file:
+
+```yaml
+apiVersion: kots.io/v1beta1
+kind: Installation
+metadata:
+  name: sentry-enterprise
+spec:
+  channelName: Stable
+  knownImages:
+  - image: docker.io/sentry-enterprise/app:1.0
+    isPrivate: true
+  releaseNotes: |
+    Release 1.0.3
+  updateCursor: "3"
+  versionLabel: 1.0.3
+  yamlErrors:
```

Comment:
it would be more clear in my opinion if we dropped a file into the errors dir that has yaml errors in it. how do we ensure the file name does not collide with a vendor yaml file?

Reply:
I'm not sure I understand this comment. Can you rephrase?

Reply:
Do you mean to include a meta file with all of the errors listed? Like `errors.yaml`? And then your concern is that `errors.yaml` could collide with a vendor file?

If so, I'd say a better UX would be to show errors in context when viewing each file.

Reply:
yes. i agree that we could make lots of ux improvements. but we also have to take into account kots pull. in this case only the file tree exists.

i suppose i could just add a file _errors.yaml which should not collide. does this file need to have a gvk? or am i overcomplicating things and should i just leave it in the installation?

Reply:
Hmm... OK. I hadn't thought of `kots pull`. Makes sense. I'll defer to your judgement on gvk.

Reply:
not gonna add a gvk